prow/crier/controller.go

```diff
@@ -165,7 +166,7 @@ func (c *Controller) updateReportState(pj *v1.ProwJob) error {
 	if newpj.Status.PrevReportStates == nil {
 		newpj.Status.PrevReportStates = map[string]v1.ProwJobState{}
 	}
-	newpj.Status.PrevReportStates[c.reporter.GetName()] = newpj.Status.State
+	newpj.Status.PrevReportStates[c.reporter.GetName()] = reportedState
```

Comment:
Before, we just set this unconditionally to `newpj.Status.State` and due to the retrying that might have changed since we did the reporting